packages/noya-state/src/reducers/pointReducer.ts

```diff
@@ -172,6 +187,133 @@ export function pointReducer(
         });
       });
     }
+    case 'setControlPointX':
+    case 'setControlPointY': {
+      if (!state.selectedControlPoint) return state;
+      const [type, amount, controlPointType, CanvasKit, mode] = action;
+      const axis = type === 'setControlPointX' ? 'x' : 'y';
+
+      const pageIndex = getCurrentPageIndex(state);
+      const layerIndexPaths = getSelectedLayerIndexPaths(state);
+      const boundingRects = getBoundingRectMap(
+        getCurrentPage(state),
+        [state.selectedControlPoint?.layerId],
+        {
+          clickThroughGroups: true,
+          includeArtboardLayers: false,
+          includeHiddenLayers: false,
+        },
+      );
+
+      return produce(state, (draft) => {
+        layerIndexPaths.forEach((indexPath) => {
+          const page = draft.sketch.pages[pageIndex];
+          const layer = Layers.access(page, indexPath);
+          const pointList = [draft.selectedControlPoint?.pointIndex];
+          const curveMode = draft.selectedControlPoint?.curveMode;
+          const boundingRect = boundingRects[layer.do_objectID];
+
+          if (!Layers.isPointsLayer(layer) || !pointList || !boundingRect)
+            return;
+
+          // Update all points by first transforming to the canvas's coordinate system
+          layer.points
+            .filter((_, index) => pointList.includes(index))
+            .forEach((curvePoint) => {
+              const decodedPoint = decodeCurvePoint(curvePoint, boundingRect);
+              if (!controlPointType) return state;
+
+              //TODO: Fix Asymmetric point values
+
+              if (
+                (controlPointType === 'curveFrom' ||
+                  controlPointType === 'curveTo') &&
+                (curveMode === 2 || curveMode === 3)
```

Comment:
We can use enums like `Sketch.CurveMode.Mirrored`, etc, instead of the numbers directly